pkg/networkservice/common/connect/DESIGN.md

```diff
@@ -19,10 +19,10 @@ return without error.
 # Implementation
 
 ## connectClient
-Each incoming server Connection must be tranlated to its corresponding client Connection for subsequent server Request and Server Close
-calls. For this reason connectClient is implemented, to manage that translation.  This results in one 'client' per server Connection.
+Each incoming server Connection must be translated to its corresponding client Connection for subsequent server Request and Server Close
```

Comment:
I'm not sure we need this step anymore now that we've fixed the Path to be correct throughout the server chain.

Reply:
> now that we've fixed the Path to be correct

To clarify: did you meant this PR or something else?

Reply:
Oh... to clarify... the whole connectClient was to map connection IDs back and forth.

Currently, the same Connection.Id is used for both server and client ends of the 'Connect' (lets call then Server1 and Client1) (and this is correct, as they share a common 'PathSegment' by design).  But the Server the Client1 is connecting to (lets call it Server2) will be different.

At the time connect was written Connection.Id inside Client1/Server1 was mixed, sometimes it was Id1, sometimes it was Id2 (from Server2).  Thus the connectClient to translate between them.  Since then, we've made it so that Id1 is used throughout Client1/Server1 (see: https://github.com/networkservicemesh/sdk/pull/470/files#diff-ab8f2b27ee7aabc91b63d5fbf418605b97002fe1d91bbc88111574517db72542 ).

Because of this, I suspect we don't need the functionality connectClient is providing here.

Reply:
Ok, got it.

I've tried to remove the translation logic from `connectClient` at all, but it resulted in a failing docker test for `cmd-forwarder-vppagent`. So, it is still required somewhere.

By Denis's suggestion, I've updated this pull request to store only `Connection.Mechanism` (which is simpler than the original PR proposal). That way the docker test passes.

Do you find this fix suitable? Some other tests that I should check?